### PR TITLE
Add hypothesis generator for Protein sequences

### PR DESCRIPTION
**Pull request by vaastav**
This PR adds a hypothesis generator for protein sequences using the 1 letter Amino Acid abbreviation as described in #3

### CODE REVIEW

**Comment by mbhall88** [comment]
I wonder if we should have some kind of separation between standard and synthetic amino acids.  

See https://github.com/biopython/biopython/blob/22df5f84fda4bf85f6c77b1f4f621c889c179a1e/Bio/Data/IUPACData.py#L10-L11 
and the comments below it for more information

**Comment by vaastav** [comment]
Maybe adding an allow_extended option is the way to go so that we can account for the non-standard amino acids. Currently, the implementation only has the standard amino acids.

Maybe, it's worthwhile to have a constant.py that can contain all of these various char sets.

**Comment by Benjamin-Lee** [comment]
@mbhall88, do you think it would make more sense to have the extended amino acids included by default (sort of like how we have ambiguous nucleotides included by default)? If so, should we also have the extended nucleotides included as well?

**Comment by mbhall88** [comment]
I think it would be best to not have them as default, but have an option `allow_extended` as @vaastav suggested. I guess as long as we make a strong point about this in the docs it should be fine to have them either in or out as default. My thoughts are that `allow_extended=False` would be the best default though.

I will ask my friend who is doing a PhD in structural bioinformatics what he thinks though.

**Comment by lafita** [comment]
Hey! In my experience proteins in sequence databases that come from genome sequencing experiments rarely contain non-standard amino acids. The only one I would recommend to consider is 'X', which is for unknown amino acids and is used frequently when the quality of the sequence is poor. In sequences that come from protein structures non-standard amino acids can be more important, so I guess the deafult will depend on your use case.

**Comment by Benjamin-Lee** [comment]
@lafita thanks for the suggestion! I think we should include `X` by default and have an option to allow generation of the extended protein alphabet noted in the docs. @vaastav do you want to implement the extended alphabet support or shall I?

**Comment by vaastav** [comment]
@Benjamin-Lee  Yea, I should be able to do it by tonight!

**Comment by vaastav** [comment]
This branch addresses issues #3 and #23

File: tests/test_protein.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+
+def test_2_mer():
+    assert minimal(protein(min_size=2)) == "AA"
+
+
+def test_2_mer_3_letter_abbrv():
+    seq = minimal(protein(single_letter_protein=False, min_size=2))
+    assert len(seq) == 6
+    assert seq == "AlaAla"
```
Comment (Benjamin-Lee) [inline tests/test_protein.py]: @lafita Are protein sequences ever represented with the three letter code? Or just single amino acids?

Comment (mbhall88) [reply tests/test_protein.py]: I've never heard of them being represented as three-letter codes.

**Pull request merged**